tests/console/validate_autoyast.pm

```diff
@@ -0,0 +1,55 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: smoke test for autoyast post-installation
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils;
+use y2logsstep;
+
+sub run {
+    my $self     = shift;
+    my $username = "bernhard";
+    $self->select_serial_terminal;
+
+    record_info('INFO', 'Check environment');
+    assert_script_run "env | grep \"SHELL=/bin/bash\"";
+    assert_script_run "env | grep \"HOME=\/root\"";
+    assert_script_run "env | grep \"OSTYPE=linux\"";
+
+    record_info('INFO', 'Check user');
+    assert_script_run "id -u $username |grep 1000";
+
+    if (not check_var('DISTRI', 'opensuse')) {
+        record_info('INFO', 'Check firewall is enabled and running');
+        assert_script_run qq{systemctl status firewalld.service | grep \"active \(running\)\"};
+        assert_script_run qq{systemctl list-unit-files | grep enabled | grep firewalld};
+
+        record_info('INFO', 'Check timezone is set correctly');
+        assert_script_run "timedatectl | grep Berlin";
+
+        record_info('INFO', 'Verify networking');
+        assert_script_run "ip link show eth0 | grep UP";
+    }
+    else {
+        # opensuse_gnome has simpler configuration and timezone or network are not defined
+        record_info('INFO', 'Check firewall is not enabled and not running');
+        assert_script_run qq{systemctl status firewalld.service | grep \"inactive \(dead\)\"};
```

Comment:
For SLE 12 we still have SUSEfirewall, you can use [firewall subroutine](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L724) to check status.